source/extensions/filters/http/ext_proc/matching_utils.cc

```diff
@@ -1,37 +1,33 @@
 #include "source/extensions/filters/http/ext_proc/matching_utils.h"
 
 #include <memory>
-#include <typeinfo>
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace ExternalProcessing {
 
-absl::flat_hash_map<std::string, std::unique_ptr<ExpressionManager::ExpressionPtrWithExpr>>
-ExpressionManager::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) const {
-  absl::flat_hash_map<std::string, std::unique_ptr<ExpressionManager::ExpressionPtrWithExpr>>
-      expressions;
+absl::flat_hash_map<std::string, Filters::Common::Expr::ExpressionPtr>
+ExpressionManager::initExpressions(const Protobuf::RepeatedPtrField<std::string>& matchers) {
+  absl::flat_hash_map<std::string, Filters::Common::Expr::ExpressionPtr> expressions;
 #if defined(USE_CEL_PARSER)
   for (const auto& matcher : matchers) {
     auto parse_status = google::api::expr::parser::Parse(matcher);
     if (!parse_status.ok()) {
       throw EnvoyException("Unable to parse descriptor expression: " +
                            parse_status.status().ToString());
     }
-    const google::api::expr::v1alpha1::Expr& parse_status_expr = parse_status.value().expr();
-    const Filters::Common::Expr::ExpressionPtr& expression =
-        Extensions::Filters::Common::Expr::createExpression(builder_->builder(), parse_status_expr);
-    std::unique_ptr<ExpressionPtrWithExpr> expr =
-        std::make_unique<ExpressionPtrWithExpr>(parse_status_expr, std::move(expression.get()));
-    if (matcher == "request.path") {
-      ExpressionManager::printExprPtrAndType(*expr, "after construction");
-    }
-    expressions.try_emplace(matcher, std::move(expr));
-    if (matcher == "request.path") {
-      ExpressionManager::printExprPtrAndType(*expressions.at(matcher),
-                                             "after placing in container");
-    }
+
+    auto iter = expr_list_.emplace_back(parse_status.value());
```

Comment:
as `emplace_back` returns a reference you need to use `auto&` or it will otherwise copy